bot/exts/utilities/colour.py

```diff
@@ -0,0 +1,230 @@
+                await self.send_colour_response(ctx, extra_colour)
+            except ValueError:
+                await invoke_help_command(ctx)
+
+    @colour.command()
+    async def rgb(self, ctx: commands.Context, red: int, green: int, blue: int) -> None:
+        """Command to create an embed from an RGB input."""
+        if any(c not in range(0, 256) for c in (red, green, blue)):
+            raise commands.BadArgument(
+                message=f"RGB values can only be from 0 to 255. User input was: `{red, green, blue}`."
+            )
+        rgb_tuple = (red, green, blue)
+        await self.send_colour_response(ctx, rgb_tuple)
+
+    @colour.command()
+    async def hsv(self, ctx: commands.Context, hue: int, saturation: int, value: int) -> None:
+        """Command to create an embed from an HSV input."""
+        if (hue not in range(0, 361)) or any(c not in range(0, 101) for c in (saturation, value)):
+            raise commands.BadArgument(
+                message="Hue can only be from 0 to 360. Saturation and Value can only be from 0 to 100. "
+                f"User input was: `{hue, saturation, value}`."
+            )
+        hsv_tuple = ImageColor.getrgb(f"hsv({hue}, {saturation}%, {value}%)")
+        await self.send_colour_response(ctx, hsv_tuple)
+
+    @colour.command()
+    async def hsl(self, ctx: commands.Context, hue: int, saturation: int, lightness: int) -> None:
+        """Command to create an embed from an HSL input."""
+        if (hue not in range(0, 361)) or any(c not in range(0, 101) for c in (saturation, lightness)):
+            raise commands.BadArgument(
+                message="Hue can only be from 0 to 360. Saturation and Lightness can only be from 0 to 100. "
+                f"User input was: `{hue, saturation, lightness}`."
+            )
+        hsl_tuple = ImageColor.getrgb(f"hsl({hue}, {saturation}%, {lightness}%)")
+        await self.send_colour_response(ctx, hsl_tuple)
+
+    @colour.command()
+    async def cmyk(self, ctx: commands.Context, cyan: int, magenta: int, yellow: int, key: int) -> None:
+        """Command to create an embed from a CMYK input."""
+        if any(c not in range(0, 101) for c in (cyan, magenta, yellow, key)):
+            raise commands.BadArgument(
+                message=f"CMYK values can only be from 0 to 100. User input was: `{cyan, magenta, yellow, key}`."
+            )
+        r = round(255 * (1 - (cyan / 100)) * (1 - (key / 100)))
+        g = round(255 * (1 - (magenta / 100)) * (1 - (key / 100)))
+        b = round(255 * (1 - (yellow / 100)) * (1 - (key / 100)))
+        await self.send_colour_response(ctx, (r, g, b))
+
+    @colour.command()
+    async def hex(self, ctx: commands.Context, hex_code: str) -> None:
+        """Command to create an embed from a HEX input."""
+        if "#" not in hex_code:
+            hex_code = f"#{hex_code}"
+        hex_tuple = ImageColor.getrgb(hex_code)
+        await self.send_colour_response(ctx, hex_tuple)
```

Comment:
```suggestion
        if "#" not in hex_code:
            hex_code = f"#{hex_code}"
        if len(hex_code) not in (4, 5, 7, 9) or any(_ not in string.hexdigits+"#" for _ in hex_code):
            raise commands.BadArgument(
                message=f"HEX values must take the form **#RRGGBB(AA)** or **#RGB(A)**. User input was: `{hex_code}`.")

        hex_tuple = ImageColor.getrgb(hex_code)
        if len(hex_tuple) == 4:
            hex_tuple = hex_tuple[:-1]  # color must be RGB. If RGBA, we remove the alpha value
        await self.send_colour_response(ctx, hex_tuple)
```
We weren't checking for the correct format from the user here. 

Should we include a message in the embed description if a hex color with alphas is given? It would tell the user that the alpha is present, but is not displayed in the color preview

Reply:
I'm not sure the current code is set up to handle RGBa hex codes. Is this something that should be added in addition? I feel like the most common is #0099FF type of hex. If there is an error message to raise, maybe we can raise it with the limits of the alphanumerics used? Like "Hex values must be 6 characters long, with characters being in the range of 0-9 and A-F."

Reply:
You're right, we can't handle alpha. I convert it to RGBA and remove the alpha if there is one. I don't think it's helpful to not convert the color just because there's an alpha included. Instead, we should convert the rest of the color and ignore the alpha, showing a message that transparency will not be shown. We could even put an indicator for transparency as a percentage, separate to the color preview

Reply:
Does the code break if an RGBA color is sent?

Reply:
> 
> 
> Does the code break if an RGBA color is sent?

yes. Specifically `discord.Color.from_rgb(*rgb)`, since it only takes 3 args

Reply:
Since it wasn't mentioned in the original issue, I am not going to implement alpha support for the base PR. The hex codes having that as an input will pass silently, but will fault out if too long.